src/main/webapp/projectContainer/Submissions.js

```diff
@@ -1,47 +1,151 @@
+    { name: 'Documents', value: 'documents' },
+    { name: 'Created', value: 'createDate' },
+  ];
+
+const styles = {
+  submissionComment: {
+    margin: '0 10px 10px 0',
+    paddingLeft: '20px',
+    width: 'calc(100% - 60px)',
+    display: 'inline-block',
+    overflow: 'visible',
+    whiteSpace: 'normal',
+    textOverflow: 'initial',
+    wordBreak: 'break-word'
+  },
+
+  addSubmission: {
+    position: 'absolute',
+    right: '30px',
+    top: '46px'
+  },
+
+  submissionCounter: {
+    marginLeft: '1em'
+  }
+};
 
 export const Submissions = hh(class Submissions extends Component {
 
   constructor(props) {
     super(props);
     this.state = {
-      content: ''
+      content: '',
+      amendments:[],
+      others: [],
+      amendmentDocuments: [],
+      otherDocuments: [],
+      submissions: {},
     };
   }
 
   componentDidMount() {
-    this.getSubmissions();
+    this.getDisplaySubmissions();
   }
 
-  getSubmissions() {
-    ProjectMigration.getSubmissions(component.projectKey).then(resp => {
+  getDocumentLink = (data) => {
+    return [component.serverURL, 'api/files-helper/get-document?id=' + data].join("/");
+  };
+
+  submissionEdit = (data) => {
+    const indexButton = a({
+      className: 'btn btn-default btn-xs pull-left link-btn',
+      href: `${component.contextPath}/submission/index?projectKey=${component.projectKey}&submissionId=${data.id}`
+    }, [component.isAdmin === true ? 'Edit': 'View']);
+    const submissionComment = span({style: styles.submissionComment}, [data.comments]);
+    return h(Fragment, {}, [indexButton, submissionComment]);
+  };
+
+  getDisplaySubmissions = () => {
+    let submissions = {};
+    ProjectMigration.getDisplaySubmissions(component.projectKey).then(resp => {
+      submissions = resp.data.groupedSubmissions;
+
+      _.map(submissions, (data, title) => {
+        data.forEach(submisionData => {
+          submisionData.documents.forEach(document => {
+            Files.getDocument(document.id).then(resp => {
+              document.document = resp.data.document;
+            });
+          });
+        });
+      });
+
       this.setState(prev => {
-        prev.content = resp.data;
+        prev.submissions = submissions;
         return prev;
-      }, () => {
-        this.loadSubmissions();
-      })
+      });
+    }).catch(error => {
+      this.setState(() => { throw error; });
     });
   };
 
-  loadSubmissions() {
-    $(".submissionTable").DataTable({
-      dom: '<"H"Tfr><"pull-right"B><div>t</div><"F"lp>',
-      buttons: [],
-      language: { search: 'Filter:' },
-      pagingType: "full_numbers",
-      pageLength: 50,
-      columnDefs: [{ targets: [1, 2], orderable: false }],
-      order: [0, "desc"]
-    });
-    $("#submission-tabs").tabs();
+  redirectNewSubmission(e) {
+    window.location.href = `${component.serverURL}/api/submissions/add-new?projectKey=${component.projectKey}&type=${e.target.id}`;
```

Comment:
Do we have to redirect here? Can we not load them? Any kind of location change like this breaks the current context.

Reply:
I think this will be solved when I complete https://broadinstitute.atlassian.net/browse/BTRX-742

Reply:
Sounds good 👍